lib/DxcSupport/WinIncludes.cpp

```diff
@@ -0,0 +1,44 @@
+//===- WinIncludes.cpp -----------------------------------------*- C++ -*-===//
```

Comment:
I know you just pulled this code out from another file, but we should clang-format it, and ideally the full patch.

(see: https://clang.llvm.org/docs/ClangFormat.html#script-for-patch-reformatting).

Reply:
I tried the latest LLVM 15 version of clang-format, and unforunately, when I use clang-format-diff.py on the patch, I get unexpected results. For instance:

```
C:\src\external\DirectXShaderCompiler>where clang-format.exe
c:\Program Files\LLVM\bin\clang-format.exe

C:\src\external\DirectXShaderCompiler>git diff -U0 --no-color --relative HEAD~1 | py -3 "c:\Program Files\LLVM\share\clang\clang-format-diff.py" -p1
--- include/dxc/Support/WinIncludes.h   (before formatting)
+++ include/dxc/Support/WinIncludes.h   (after formatting)
@@ -125,12 +125,12 @@
 #ifndef _WIN32
 CROSS_PLATFORM_UUIDOF(IMalloc, "00000002-0000-0000-C000-000000000046")
 struct IMalloc : public IUnknown {
-  virtual void *Alloc(SIZE_T size) = 0;
-  virtual void *Realloc(void *ptr, SIZE_T size) = 0;
-  virtual void Free(void *ptr) = 0;
-  virtual SIZE_T GetSize(void *pv) = 0;
-  virtual int DidAlloc(void *pv) = 0;
-  virtual void HeapMinimize(void) = 0;
+        virtual void *Alloc(SIZE_T size) = 0;
+        virtual void *Realloc(void *ptr, SIZE_T size) = 0;
+        virtual void Free(void *ptr) = 0;
+        virtual SIZE_T GetSize(void *pv) = 0;
+        virtual int DidAlloc(void *pv) = 0;
+        virtual void HeapMinimize(void) = 0;
 };
 #endif
```

Note how the diff indented the member functions in `IMalloc` with 8 spaces. The `.clang-format` file sets the style to LLVM, which uses 2 space indentation. If I run clang-format directly against this file, it indeed spits out this class with 2-space indented functions as expected:

```
C:\src\external\DirectXShaderCompiler>clang-format C:\src\external\DirectXShaderCompiler\include\dxc\Support\WinIncludes.h

<SNIP>

#ifndef _WIN32
CROSS_PLATFORM_UUIDOF(IMalloc, "00000002-0000-0000-C000-000000000046")
struct IMalloc : public IUnknown {
  virtual void *Alloc(SIZE_T size) = 0;
  virtual void *Realloc(void *ptr, SIZE_T size) = 0;
  virtual void Free(void *ptr) = 0;
  virtual SIZE_T GetSize(void *pv) = 0;
  virtual int DidAlloc(void *pv) = 0;
  virtual void HeapMinimize(void) = 0;
};
#endif

HRESULT DxcCoGetMalloc(DWORD dwMemContext, IMalloc **ppMalloc);

#endif // defined(_WIN32) && !defined(DXC_DISABLE_ALLOCATOR_OVERRIDES)
```

I'm not sure what's going on. Maybe clang-format doesn't work properly without the full context fo the file? Should I bother with this? Other files also now have 8 space indentation.

Reply:
To add to what I wrote above, attempting to fully format almost any file in the codebase seems to produce a significant difference. I think if the goal is to clang-format the entire codebase, this should perhaps be done first, and then enforced with pre-push hooks and CI. If it's alright, I'd like to land this as-is.

There seems to be one more "workflow awaiting approval", but I'm nto sure what that means:

![image](https://github.com/microsoft/DirectXShaderCompiler/assets/6893883/46ec1e0b-566e-4d40-a415-e938d7c93e77)

Reply:
Our codebase is not clang-format clean, but new code introduced should be clang-formatted. If `clang-format-diff` isn't finding the format settings in the root of the repo that seems like a configuration issue. It has worked for me without issue. `clang-format-diff` does work without the full context, so this seems like a configuration problem. Beyond that I don't really have guidance.

Reply:
Oops, pasted my comment in the wrong place:

I don't believe it's a configuration issue. Maybe you can try it with my
patch?